src/machines/sketchSolveMode.ts

```diff
@@ -0,0 +1,136 @@
+    }),
+    'send updated selection to move tool': sendTo('moveTool', {
+      type: 'update selection',
+    }),
+    'request unequip': sendTo('toolInvoker', {
+      type: 'unequip',
+    }),
+  },
+  actors: {
+    moveToolActor: createMachine({
+      /* ... */
+    }),
+    toolInvoker: (({ input }: { input: { tool: EquipTool } }) =>
+      getToolMachine(input.tool)) as unknown as AnyActorLogic,
+  },
+}).createMachine({
+  /** @xstate-layout N4IgpgJg5mDOIC5QGUDWYAuBjAFgAmQHsAbANzDwFlCIwBiMADwEsMBtABgF1FQAHQrFbNCAO14hGiAIwAmAKwAOAHQcAzABZFGgGwBOebL16A7LIA0IAJ6INencs0a7ixTvUdpAXy+W0mXAIScioaegBXPggAQwwKWDBiMCwMEVFOHiQQASFUsQkpBHk1RwV7WTVFWVlFaR0dSxsEAFpqjmVZXT0OWXcTeQMenz90bHwiMgpqWjpw0TAAR3DmPjwMQhIMiRzhfKzCkx15ZSOOHu61fpNGxBM1dqOdGsNXRRNpRWGQfzGgydDaMoALaEELRUQQPAJJIpOgQMRgZTMUSkQjoYGgsAAFQ2xC2WR2eXE+xkFWkylc-SMT3kdWkNwQ1RKHHebxqeiqtXk8i+P0CExC00RILBEKhiWSGAYACdpYRpco+MRYgAzeVAjHkHGbbjbQS7YmgQpyNTkymGcq0nT06wyEwmDo1OQmDTSV33T6+b6jfnBKZhTUUcGQ6GShhLFZrXH4-j6okFWwsxzSHpPWQmDnp+QM6TydqKDg6S5FgxKRRqXk+8Z+gGI8JCURQKMkOEIpEotGI9YkACSHfQ0pj2TjaQTRSLymMlxM5cLprU2dtjOtJw+h08edddUrAWr-yFynryKb3eIMrlCqVqvVylPfdRA6HhNHJPHJSn9tnRekC5zpmUdg6G47xqO40hmBoPheqIYTwFkfJ7oKYR6rkL5Gog6b5pc8gaD0dzdB8DJqHojq1J0Sg1JUrg7r8Ar+oCIpBmKoYpChBpjncJRqAuejEQMhxcjmvTKGYZHaBwWgKP0NG+vuAZHo2zbEGx8avqaKjlv0uHpsRniKERSgAWcP6KHosi5jOnojLufxIYCTDCI2KloZIMjESoroKO5FRFn+Dr1G8ubyGYOFPBWUFAA */
+  context: ({ input }): ModelingMachineContext => ({
+    ...modelingMachineDefaultContext,
+  }),
+  id: 'Sketch Solve Mode',
+  initial: 'move and select',
+  on: {
+    exit: {
+      target: '#Sketch Solve Mode.exiting',
+      actions: 'send unequip to tool',
+      description:
+        'the outside world can request that sketch mode exit, but it needs to handle its own teardown first.',
+    },
+    'update selection': {
+      actions: [
+        'send update selection to equipped tool',
+        'send updated selection to move tool',
+      ],
+      description:
+        'sketch mode consumes the current selection from its source of truth (currently modelingMachine). Whenever it receives',
+    },
+    'unequip tool': {
+      actions: 'request unequip',
+    },
+  },
+  states: {
+    'move and select': {
+      entry: () => console.log('entered sketch mode'),
+      on: {
+        'equip tool': {
+          target: 'using tool',
+        },
+      },
+      invoke: {
+        id: 'moveTool',
+        input: {},
+        onDone: {
+          target: 'exiting',
+        },
+        onError: {
+          target: 'exiting',
+        },
+        src: 'moveToolActor',
+      },
+      description:
+        'The base state of sketch mode is to all the user to move around the scene and select geometry.',
+    },
+    'using tool': {
+      invoke: {
+        id: 'toolInvoker',
+        input: ({
+          event,
+        }: { event: SketchSolveMachineEvent }): { tool: EquipTool } => ({
+          tool: event.type === 'equip tool' ? event.data.tool : 'dimension',
```

Comment:
This defaults to "dimension" while the implementation of `getToolMachine` falls back to "center rectangle", seems conflicting

Reply:
that's fair, not sure ethier of these are great choices though. but made them consistent.